k9mail/src/main/java/com/fsck/k9/fragment/TagChoiceDialogFragment.java

```diff
@@ -0,0 +1,165 @@
+                listener.onStartKeywordEditor();
+                dismiss();
+            }
+        });
+
+        tags = Keyword.getVisibleKeywords();
+        TagListAdapter adapter = new TagListAdapter(getActivity(), tags);
+        ListView listView = (ListView) tagChoiceListView.findViewById(
+            R.id.tag_choice_list_view);
+
+        listView.setAdapter(adapter);
+        listView.setChoiceMode(ListView.CHOICE_MODE_MULTIPLE);
+        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(
+                AdapterView parent, View view, int position, long id)
+            {
+                ListView lv = (ListView) parent;
+                Keyword clickedTag = tags.get(position);
+                if (lv.isItemChecked(position)) {
+                    selectedTags.add(clickedTag);
+                } else {
+                    selectedTags.remove(clickedTag);
+                }
+            }
+        });
+
+        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
+        builder.setTitle(R.string.tag_choice_dialog_title)
+            .setView(tagChoiceListView)
+            .setPositiveButton(
+                android.R.string.ok, new DialogInterface.OnClickListener()
+            {
+                @Override
+                public void onClick(DialogInterface dialog, int id) {
+                    Set<Keyword> addedTags = new HashSet<Keyword>(selectedTags);
+                    addedTags.removeAll(preSelectedTags);
+                    Set<Keyword> deletedTags = new HashSet<Keyword>(preSelectedTags);
+                    deletedTags.removeAll(selectedTags);
+                    listener.onStoreTags(addedTags, deletedTags);
+                }
+            })
+            .setNegativeButton(android.R.string.cancel, null);
+
+        return builder.create();
+    }
+
+    @Override
+    public void onAttach(Activity activity) {
+        super.onAttach(activity);
+        try {
+            listener = (TagChoiceDialogListener) activity;
+        } catch (ClassCastException e) {
+            throw new ClassCastException(activity.toString() +
+                "does not implement TagChoiceDialogListener");
+        }
+    }
+
+
+    public class TagListAdapter extends ArrayAdapter<Keyword> {
+
+        TagListAdapter(Context context, List<Keyword> tags) {
+            super(context, 0 /* resource not used */, tags);
+        }
+
+        @Override
+        public View getView(int position, View convertView, ViewGroup parent) {
+            if (convertView == null) {
+                LayoutInflater inflater = getActivity().getLayoutInflater();
+                convertView = inflater.inflate(
+                    android.R.layout.simple_list_item_multiple_choice, null);
+            }
+
+            Keyword tag = getItem(position);
+            TextView tv = (TextView) convertView.findViewById(
+                android.R.id.text1);
```

Comment:
`android.R.id.text1` comes from the built-in layout `android.R.layout.simple_list_item_multiple_choice`. So, to change the former (for example to something like `keyword_name`), I would need to add a local layout for the list item. I will happily change it! Could you please confirm that this approach is preferred over using the built-in layout?